Macos:  Use pkgbuild/productbuild  to create a .pkg file

leamas opened the pull request:
An alternative way to create a .pkg file not depending on the  hard-to-maintain pkgproj file. It uses the tools pkgbuild and productbuild which basically just looks at *build/OpenCPN.app* and copies the complete structure to the package.

It does add a small configuration file distro.plist. This should not be that hard to maintain.

Example package is available at https://dl.cloudsmith.io/public/alec-leamas/opencpn/raw/files/OpenCPN_5.7.1+4990.be7881376.pkg

It does have the known problems using  homebrew binary libraries on 10.15; the libraries are linked against 12.0.

The generated package is pretty bare-bones. productbuild has a multitude of options for license dialogs, backgrund images,  title, etc. Some of these should probably be used if walking this path.

Package must be built on 14.2 since the option `--min-os-version` fed to pkgbuild only exists there. On previous versions pkgbuild generates a package which requires macos 12.0, probably after introspecting the bundled libraries. That is, there is chance we could build on earlier Xcode versions if the libraries' 10.15 linkage was sane.

bdbcat commented:
I would like to see the PKG artifact built by this new procedure be named differently in Cloudsmith upload step, so that we could retain both methods for a little while.
Possible?

leamas commented:
No problem at all, see CMakeLists.txt around line 2945, the definition of MACOS_PKG

leamas commented:
Here are lot's of rough corners. One is that the OpenCPN.app.pkg is uploaded. It is just a temporary, and should probably be removed in the last steps.

leamas commented:
hm... I see this in the log:
```
pkgbuild: unrecognized option `--min-os-version'
```

This means that the job does not run on xcode 14.2.0 , see my top comment

bdbcat commented:
"That is, there is chance we could build on earlier Xcode versions if the libraries' 10.15 linkage was sane."
I guess you are referring to things like this:
ld: warning: dylib (/usr/local/opt/openssl@3/lib/libcrypto.3.dylib) was built for newer macOS version (12.0) than being linked (10.10)
?

leamas commented:
> I guess you are referring to things like this:

Yes

leamas commented:
> So it will by definition be linked relative to the OS version running on the build host.

Unless the build tools respect MACOSX_DEPLOYMENT_TARGET.  Or we  just provide a pre-compiled bundle. Not nice, but supporting many versions backwards has a price.

leamas commented:
The circleci build is ok, but the GA needs some polish. Working on it.

leamas commented:
Rebased to current master and modified:
  - The build on and for MacOS 11 is done using the "new" way whereas the 10.15 package is using  the "old" tools.
  - The 11 package has a specific filename, different from the 10.15 package.
  - The "new" package has got some polish like a better defined version + dialog title and background.
  - The handling of the mpg123 package is changed based on forcing the cmake _find_library()_ actually find the correct version and then trust the standard workflow. 

There are still problems with the linkage level, bit I'm diverging to much and need to land this first.
